src/coord/src/catalog/storage.rs

```diff
@@ -417,6 +434,25 @@ impl Transaction<'_> {
         }
     }
 
+    pub fn load_timestamps(
+        &self,
+        source_id: GlobalId,
+    ) -> Result<Vec<(PartitionId, Timestamp, MzOffset)>, Error> {
+        self.inner
+            .prepare_cached(
+                "SELECT pid, timestamp, offset from timestamps where sid = ? order by pid, timestamp")?
+            .query_and_then(params![SqlVal(&source_id)], |row| -> Result<_, Error> {
+                let partition: PartitionId = row.get::<_, String>(0)?.parse().expect("parsing partition id from string cannot fail");
```

Comment:
is this `expect` legit? my rationale was "we wrote this down, we should expect to be able to read it back?" but im open to thoughts